packages/block-editor/src/components/media-edit/index.native.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import {
+	requestMediaPicker,
+	requestMediaEditor,
+	mediaSources,
+} from 'react-native-gutenberg-bridge';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Picker } from '@wordpress/components';
+
+export const MEDIA_TYPE_IMAGE = 'image';
+
+const editOption = {
+	id: mediaSources.mediaEditor,
+	value: mediaSources.mediaEditor,
+	label: __( 'Edit' ),
+	types: [ MEDIA_TYPE_IMAGE ],
+	icon: 'admin-appearance',
+};
+
+const replaceOption = {
+	id: mediaSources.deviceLibrary,
+	value: mediaSources.deviceLibrary,
+	label: __( 'Replace' ),
+	types: [ MEDIA_TYPE_IMAGE ],
+	icon: 'update',
+};
+
+const options = [ editOption, replaceOption ];
+
+export class MediaEdit extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.onPickerPresent = this.onPickerPresent.bind( this );
+		this.onPickerSelect = this.onPickerSelect.bind( this );
+	}
+
+	getMediaOptionsItems() {
+		return options;
+	}
+
+	onPickerPresent() {
+		if ( this.picker ) {
+			this.picker.presentPicker();
+		}
+	}
+
+	onPickerSelect( value ) {
+		const { allowedTypes = [], onSelect, multiple = false } = this.props;
+		const mediaSource = this.getMediaOptionsItems().filter( ( source ) => source.value === value ).shift();
+		const types = allowedTypes.filter( ( type ) => mediaSource.types.includes( type ) );
+		const mediaCallback = ( media ) => {
+			if ( ( multiple && media ) || ( media && media.id ) ) {
+				onSelect( media );
+			}
+		};
+
+		switch ( value ) {
+			case mediaSources.mediaEditor:
+				requestMediaEditor( this.props.source.uri, mediaCallback );
+				break;
+			default:
+				requestMediaPicker( mediaSource.id, types, multiple, mediaCallback );
```

Comment:
The `Replace` option will automatically call the Device Library picker. There is no chance to pick an image from the WP Media Library or using the camera.

What is the expectation?
cc @iamthomasbishop

Reply:
> The Replace option will automatically call the Device Library picker. There is no chance to pick an image from the WP Media Library or using the camera

The expectation would be that `replace` should allow _all_ available image sources.

Reply:
@etoledom @iamthomasbishop so, I'm assuming that when the user taps "Replace" we show another Picker, right?

Reply:
> I'm assuming that when the user taps "Replace" we show another Picker, right?

When you tap `Replace`, we'd show an action/bottom sheet that lets you pick from various image sources (wp media, device, free photo library, etc.). Once you pick a source, we launch the appropriate picker/UI.

Reply:
@iamthomasbishop Oh sorry, I meant to say Action Sheet. So, you tap the button and:

* We show the Action Sheet
* If the user taps "Replace" we show another Action Sheet with the various image sources
* Once the user taps one option we launch the picker/UI

Reply:
Exactly right 😄

Reply:
Done!

![2020-01-22 16 29 23](https://user-images.githubusercontent.com/7040243/72927163-73e1e800-3d34-11ea-88cc-7e69a353d72a.gif)